application/common/components/adapters/WorkdayIdStore.php

```diff
@@ -0,0 +1,143 @@
+<?php
+namespace Sil\Idp\IdSync\common\components\adapters;
+
+use Exception;
+use GuzzleHttp\Client;
+use InvalidArgumentException;
+use Sil\Idp\IdSync\common\components\IdStoreBase;
+use Sil\Idp\IdSync\common\models\User;
+use yii\helpers\Json;
+
+class WorkdayIdStore extends IdStoreBase
+{
+    public $apiUrl = null;
+    public $username = null;
+    public $password = null;
+    
+    public $timeout = 45; // Timeout in seconds (per call to ID Store API).
+    
+    protected $httpClient = null;
+    
+    public function init()
+    {
+        $requiredProperties = [
+            'apiUrl',
+            'username',
+            'password',
+        ];
+        foreach ($requiredProperties as $requiredProperty) {
+            if (empty($this->$requiredProperty)) {
+                throw new InvalidArgumentException(sprintf(
+                    'No %s was provided.',
+                    $requiredProperty
+                ), 1532982562);
+            }
+        }
+        
+        parent::init();
+    }
+    
+    public static function getIdBrokerFieldNames()
+    {
+        return [
+            'Employee_Number' => User::EMPLOYEE_ID,
+            'First_Name' => User::FIRST_NAME,
+            'Last_Name' => User::LAST_NAME,
+            'Display_Name' => User::DISPLAY_NAME,
+            'Email' => User::EMAIL,
+            'Username' => User::USERNAME,
+            'Account_Locked__Disabled_or_Expired' => User::LOCKED,
```

Comment:
I noticed the extra "_" before Disabled. Just wanted to point that out in case it wasn't intended.

Reply:
Thanks for reading that closely :-)
It actually is correct, though. I think the field name was generated from a more user-friendly text field.